replica/table.cc

```diff
@@ -2160,19 +2163,23 @@ future<> tablet_storage_group_manager::update_effective_replication_map(const lo
             _storage_groups[tid.value()] = std::make_unique<storage_group>(std::move(cg), &_compaction_groups);
         }
     }
-    co_return;
+    // For correctness, new erm is applied atomically with changes in storage state (before any yielding point).
+    apply_erm();
+
+    return ret;
```

Comment:
Shouldn't we wait first? I don't understand what's going on, we handle the split in parallel to everything else?

Reply:
handle_tablet_split_completion() must happen atomically with apply_erm(). handle_tablet_split_completion() is only futurized because there are empty compaction groups that had their data split into left and right compaction groups. previously, handle_tablet_split_completion() was not futurized and simply stopped those empty groups in background. 

so what we're doing here is to call handle_tablet_split_completion(). we know by the time it returns, the groups were already remapped into the new set, and then erm is applied. you cannot change the set, without applying new erm, and also the other way around (of course, atomically). otherwise new reads that start in between will have incorrect behavior.

Reply:
that's why my first version decided to split execution and waiting of background work.

Reply:
futurization of handle_tablet_split_completion() was done here f2ff7014899

Reply:
maybe it's less confusing if we change signature to be more explicit, as in seastar, we're used to thinking a function has only completed once the future is consumed. will change that.